aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Orchestrator.java

```diff
@@ -135,10 +138,16 @@ public boolean isStopped() {
      * Start performing sync operations between the local storage adapter
      * and the remote GraphQL endpoint.
      */
-    public void start() {
+    public synchronized void start() {
+        LOG.debug("Available permits = " + startStopSemaphore.availablePermits());
+        if (!startStopSemaphore.tryAcquire()) {
+            LOG.warn("Unable to acquire orchestrator lock. Transition currently in progress.");
+            return;
+        }
         disposables.add(transitionCompletable()
             .subscribeOn(startStopScheduler)
             .doOnDispose(() -> LOG.debug("Orchestrator disposed a transition."))
+            .doFinally(startStopSemaphore::release)
```

Comment:
in the problem description here: https://github.com/aws-amplify/amplify-android/issues/853 you mentioned that we should only **We truly only need to block until the storage observer starts**, in here it looks like we are still blocking everything , shall we release the ```startStopSemaphore``` as soon as ```startObservingStorageChanges()``` is done? something like:

```
  mutationOutbox.load()
                .andThen(Completable.create(emitter -> {
                    storageObserver.startObservingStorageChanges(emitter::onComplete);
                    currentMode.set(Mode.LOCAL_ONLY);
                }))
.foFinally(startStopSemaphore::release)
.blockingAwait(OP_TIMEOUT_SECONDS, TimeUnit.SECONDS);

```